src/main/java/com/pusher/client/channel/PusherEvent.java

```diff
@@ -0,0 +1,47 @@
+package com.pusher.client.channel;
+
+import java.util.Map;
+
+public class PusherEvent {
+    private Map<String, Object> eventData;
+
+    /**
+     * PusherEvent constructor that takes a hashmap.
+     * @param eventData hashmap(string, string)
+     */
+    public PusherEvent(Map<String, Object> eventData) {
+        this.eventData = eventData;
+    }
+    /**
+     * getProperty returns the value associated with the key, or null.
+     * @param key - the key you wish to get
+     * @return value (string) - the value as a string.
+     */
+    public Object getProperty(String key) {
+        return eventData.get(key);
+    }
+
+
+    /**
+     * returns the userId associated with this event.
+     * @return the userID string: https://pusher.com/docs/channels/using_channels/events#user-id-in-client-events,
+     * or null if the customer if the event is not a client event.
+     */
+    public Object getUserId() {
+        return eventData.get("user_id");
+    }
+
+    public Object getChannel() { return eventData.get("channel"); }
+    public Object getEventName() { return eventData.get("event"); }
```

Comment:
```suggestion
    public Object getName() { return eventData.get("event"); }
```

I'm not sure about this one. It feels redundant prefixing this with `Event` when the object is already a `PusherEvent`. It's also more consistent with the swift SDK: https://github.com/pusher/pusher-websocket-swift/pull/225/files#diff-91b6e8bdc0beb7f39153a182dcfbf19fR9.

The downside is the property in the JSON object is called `event`, so it might be a but confusing if it's called "name" on this object. Do you have any thoughts @jameshfisher?

Reply:
I think `getEventName` is more explicit, and I would stick to it. We didn't merge Swift PR yet and can change it, if we decide the opposite, i guess? Although I don't have a strong opinion and i guess it's up to @jameshfisher to decide.

Reply:
I agree we should be consistent, and that this is important for this public API.

I like the principle of using pusher-js as the canonical SDK. pusher-js doesn't really expose a name for this, but [it does call this `event_name` in the source and/or TypeScript bindings.](https://github.com/pusher/pusher-js/blob/master/src/core/pusher.ts#L192)

Based on that, I prefer `getEventName`, and renaming the property in Swift to `eventName`. cc @TomKemp.

(I would also weakly prefer `getChannelName` as opposed to `getChannel`, to clarify the difference between `String` and the `Channel` class. But this is a different discussion.)

Reply:
Fair enough, although I agree that if we are calling this `getEventName`, then we should call `getChannel` `getChannelName`.

(it might be a different discussion, but it's important we have it now).

Reply:
It's exposed as `channel_name` in auth endpoint handlers for example: https://pusher.com/docs/channels/server_api/authenticating-users#implementing-the-auth-endpoint-for-a-private-channel.

Reply:
I agree with `getChannelName`, happy to change.